lib/Agrammon/Web/APIRoutes.pm6

```diff
@@ -28,14 +26,64 @@ my class AgrammonAPITokenMiddleware does Cro::APIToken::Middleware {
 }
 
 sub api-routes(Agrammon::Web::Service $ws) is export {
+    my $schema = 'share/agrammon-rest.openapi';
     route {
+        if %*ENV<AGRAMMON_DEBUG> {
+            before {
+                # Consume and re-instate request.
+                my $blob = await request.body-blob;
+                request.set-body($blob);
+                # Dump.
+                my $req = ~request;
+                try $req ~= $blob.decode('utf-8');
+                note "request=$req";
+            }
+        }
         before AgrammonAPITokenMiddleware.new(manager => get-api-token-manager());
 
-        get -> APIUser $user, 'greet' {
-            content 'application/json', { message => "Hello $user.firstname()" }
-        }
+        include rest-api-routes($schema, $ws);
     }
 }
 
+sub rest-api-routes (Str $schema, Agrammon::Web::Service $ws) is export {
+    openapi $schema.IO, {
+        operation 'greetUser', -> APIUser $user {
+            content 'application/json', { message => "Hello $user.firstname()" }
+        }
+        operation 'getLatex', -> APIUser $user, :$technical = 'technical.cfg', :$sort = 'model' {
+            content 'text/plain', $ws.get-latex($technical, $sort)
+        }
 
+        operation 'getTechnical', -> APIUser $user, :$technical = 'technical.cfg' {
+            content 'text/plain', $ws.get-technical($technical);
+        }
 
+        operation 'runSimulation', -> APIUser $user {
+            request-body 'multipart/form-data' => -> (
+                :$simulation!,  :$dataset!, :$inputs!, :$technical='',
+                :$model = 'version6', :$variants = 'Base', :$language = 'de', :$format = 'text/plain',
+                :$print-only = '', :$include-filters = 'false', :$all-filters = 'false'
+            ) {
+                my $type = $inputs.content-type;
+                if $type ne 'text/csv' {
+                    my $error = "Content type is '$type', must be 'text/csv'";
+                    note $error;
+                    bad-request 'application/json', %( error => $error );
+                }
+                else {
+                    my $input-data = $inputs.body-text;
+                    my $results = $ws.get-outputs-from-csv(
+                        $user,
+                        ~$simulation, ~$dataset, ~$input-data,
+                        :model-version(~$model), :variants(~$variants), :technical-file(~$technical),
+                        :language(~$language), :format(~$format), :print-only(~$print-only),
```

Comment:
If we were being really well behaved here in terms of HTTP we'd take the format in the `Accept` header.

Reply:
Makes sense, done.